Whisky/Utils/Logger.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  Logger.swift
+//  Whisky
+//
+//  Created by Isaac Marovitz on 08/06/2023.
+//
+
+import Foundation
+import OSLog
+
+class Log {
+    static let logsFolder = FileManager.default.urls(for: .libraryDirectory,
+                                                    in: .userDomainMask)[0]
+        .appendingPathComponent("Logs")
+        .appendingPathComponent("Whisky")
+
+    let fileHandle: FileHandle
+    let logger: Logger
+
+    init?() {
+        do {
+            if !FileManager.default.fileExists(atPath: Log.logsFolder.path) {
+                try FileManager.default.createDirectory(at: Log.logsFolder, withIntermediateDirectories: true)
+            }
+
+            let formatter = DateFormatter()
+            formatter.dateFormat = "d MMM y HH-mm"
+
+            let dateString = formatter.string(from: Date.now)
```

Comment:
Instead of calling out to `Date.now`, add `date: () -> Date = Date.init` to your initialiser, and call it here: `date()`
This makes the Logger a bit more testable, as you can pass in any `Date` instance you want then.